### ISSUE TITLE
Work-around for explicit DAE system?

### ISSUE THREAD

**Issue by brunomorampc** [patrick-kidger/diffrax#413]
Imagine I have 

$$\dot{x}= f(x,g(x))$$

I want the diffeqsolve to return both  the state and $g(x)$ at the different time points. I know from #62 that the DAE are not yet implemented, but is there a work-around that I can implement to output $g(x)$ together with the state?

**Comment by patrick-kidger** [comment]
Can you evaluate `g` on the outputs of `diffeqsolve`?

**Comment by brunomorampc** [comment]
Yes, true.  I did not write the complete story to avoid making it complicated but it turned out to be misleading.

I was thinking to express this as a DAE by defining like z = g(x) such that I can get that as an output of the ODE because the problem is that I do not want to re-construct the `g` function outside of  the diffeqsolve to avoid complications in my code because that function is farely complicated and can change from applications to applications.

**Comment by patrick-kidger** [comment]
I don't think there's a way to do this in general I'm afraid. This isn't just a limitation of Diffrax, it's part of the underlying numerical methods: the points at which we evaluate the vector field are not necessarily the same as the points at which we return outputs.
